src/app/handlers/new_member_role.handler.ts

```diff
@@ -7,6 +7,12 @@ export class NewMemberRoleHandler implements IHandler {
   constructor(public container: IContainer) {}
 
   public async execute(member: GuildMember): Promise<void> {
+    const persistedRoles = this.container.userService.getPersistedRoles(member.id);
+    if (persistedRoles) {
+      await member.roles.add(persistedRoles);
+      return; // No need to check the rest if they were already in the server
+    }
```

Comment:
sketchy to have this not be in its own handler

Reply:
ok, will do